src/chat/fetchingReducer.js

```diff
@@ -67,6 +68,12 @@ export default (
 ): FetchingState => {
   switch (action.type) {
     case RESET_ACCOUNT_DATA:
+
+    // Reset just because `fetching` is server-data metadata, and we're
+    // resetting the server data it's supposed to apply to… but really, if
+    // there's an in-progress message fetch now, it's a bug, perhaps #5609.
+    // TODO(#5609): Remove "perhaps #5609" when that specific bug is fixed.
+    case REGISTER_COMPLETE: // eslint-disable-line no-fallthrough
```

Comment:
As discussed at https://github.com/zulip/zulip-mobile/issues/5609#issuecomment-1353905491 just now, I think it's actually fine for there to still be an ongoing fetch up to this point.

There's still the issue that even though we reset this state to *say* there's no longer an ongoing fetch, there may still be an ongoing fetch even after this point. That's #5609. But even if there is one, I think we probably don't want to be showing loading indicators for it in the UI, so it's right to be resetting `state.fetching` here.

Reply:
When we do cancel the fetches, we can do a `MESSAGE_FETCH_ABORT` or some-such action so that `state.fetching` should be all `false` when `REGISTER_COMPLETE` comes along, right, what do you think?

That way it's more transparent that a given fetch has `false` in `state.fetching` because we aborted that fetch.

Reply:
Sure, that seems reasonable. We'll see what feels cleanest when we get to that point, I guess.